### PR TITLE
[Mobile] - React Native 0.69.4 Upgrade - Android

### PR DESCRIPTION
**Pull request by geriux**
### Feature branch

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5193)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43485)

### Android integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5170)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43486)
- [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android/pull/17066)

### iOS integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5192)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/44889)
- [WordPress iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/19326)

## What?
This PR includes the needed Android changes to upgrade the React Native framework to the `0.69.4` version.

## Why?

To keep up to date with the latest stable React Native version.

## How?

Using the React Native Upgrade Helper ([list of changes](https://react-native-community.github.io/upgrade-helper/?from=0.66.2&to=0.69.4)):

- It adds all of the new files to support the new React Native architecture, files under the `gutenberg/newarchitecture` and  `main/jni` all of those were copied with no changes other than the package name `package com.gutenberg.newarchitecture;`
- It updates the usage of `GestureHandlerRootView` since it is now deprecated to do it in the `MainActivity`, so it was removed from [here](https://github.com/WordPress/gutenberg/pull/43486/files#diff-769716bb706f66f37212cb2c4b06ec2e87280a37c66d795ad8b2ca56843e7a95) and added in the [javascript side](https://github.com/WordPress/gutenberg/pull/43486/files#diff-d45b169f7a1824d1f08790b97680e012bb30be66ed5714994a3959ba23cdee94R178-R190).
- It [updates](https://github.com/WordPress/gutenberg/pull/43486/files#diff-1f256578dd9ab8dda6c3913a55a588312129f3404bb0cdf0b1ab52f0a6937119R92) the path for `hermesc` since it is now under the React Native folder.

**Before merging:**
- [x] Update the third-party dependencies for React Native Video, React Native Reanimated and React Native Gesture Handler.

## Testing Instructions
The Android app should build and open the editor correctly.

## Screenshots or screencast 
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -243,7 +243,7 @@
 	"scripts": {
 		"build": "npm run build:packages && wp-scripts build",
 		"build:analyze-bundles": "npm run build -- --webpack-bundle-analyzer",
-		"build:package-types": "node ./bin/packages/validate-typescript-version.js && tsc --build",
+		"build:package-types": "node ./bin/packages/validate-typescript-version.js",
```
Comment (geriux) [inline package.json]: We should revert this before merging

File: packages/react-native-bridge/android/react-native-bridge/build.gradle [hunk 2]
```diff
@@ -65,25 +70,30 @@ dependencies {
 
     implementation "com.facebook.react:react-native:$rnVersion"
     implementation "org.wordpress-mobile:react-native-svg:${extractPackageVersion(packageJson, 'react-native-svg', 'dependencies')}"
-    implementation "com.github.wordpress-mobile:react-native-video:${extractPackageVersion(packageJson, 'react-native-video', 'dependencies')}"
+    // implementation "com.github.wordpress-mobile:react-native-video:${extractPackageVersion(packageJson, 'react-native-video', 'dependencies')}"
+    implementation "com.github.wordpress-mobile:react-native-video:update-for-0.69-SNAPSHOT"
```
Comment (geriux) [inline packages/react-native-bridge/android/react-native-bridge/build.gradle]: This needs to be updated once we released the new version of React Native Video.

File: packages/react-native-bridge/android/react-native-bridge/build.gradle [hunk 3]
```diff
@@ -65,25 +70,30 @@ dependencies {
...
     implementation "com.github.wordpress-mobile:react-native-slider:${extractPackageVersion(packageJson, '@react-native-community/slider', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-get-random-values:${extractPackageVersion(packageJson, 'react-native-get-random-values', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-masked-view:${extractPackageVersion(packageJson, '@react-native-masked-view/masked-view', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-screens:${extractPackageVersion(packageJson, 'react-native-screens', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-safe-area-context:${extractPackageVersion(packageJson, 'react-native-safe-area-context', 'dependencies')}"
-    implementation "com.github.wordpress-mobile:react-native-reanimated:${extractPackageVersion(packageJson, 'react-native-reanimated', 'dependencies')}"
+    //implementation "com.github.wordpress-mobile:react-native-reanimated:${extractPackageVersion(packageJson, 'react-native-reanimated', 'dependencies')}"
+    implementation "com.github.wordpress-mobile:react-native-reanimated:wp-fork-2.9.1-SNAPSHOT"
```
Comment (geriux) [inline packages/react-native-bridge/android/react-native-bridge/build.gradle]: This needs to be updated once we released the new version of React Native Reanimated.

File: packages/react-native-bridge/android/react-native-bridge/build.gradle [hunk 4]
```diff
@@ -65,25 +70,30 @@ dependencies {
...
     implementation "com.github.wordpress-mobile:react-native-prompt-android:${extractPackageVersion(packageJson, 'react-native-prompt-android', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-webview:${extractPackageVersion(packageJson, 'react-native-webview', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-clipboard:${extractPackageVersion(packageJson, '@react-native-clipboard/clipboard', 'dependencies')}"
     implementation "org.wordpress-mobile:react-native-fast-image:${extractPackageVersion(packageJson, 'react-native-fast-image', 'dependencies')}"
 
-    implementation("com.github.wordpress-mobile:react-native-gesture-handler:${extractPackageVersion(packageJson, 'react-native-gesture-handler', 'dependencies')}", {
+    // implementation("com.github.wordpress-mobile:react-native-gesture-handler:${extractPackageVersion(packageJson, 'react-native-gesture-handler', 'dependencies')}", {
+    implementation("com.github.wordpress-mobile:react-native-gesture-handler:wp-fork-2.5.0-SNAPSHOT", {
```
Comment (geriux) [inline packages/react-native-bridge/android/react-native-bridge/build.gradle]: This needs to be updated once we released the new version of React Native Gesture Handler.

**Review by derekblank — APPROVED** [review]
LGTM, pending the updates to react-native-reanimated and react-native-gesture-handler. 🚀

**Pull request merged**